### PR TITLE
Feat(orchestrator): add substrate cli args version

### PR DESCRIPTION
**Pull request by pepoviola**
- Add logic to manage different versions of `substrate cli arguments` (needed since this breaking change https://github.com/paritytech/substrate/pull/13384). 

At the moment will be only have 2 versions and the main change is that `ws-port` is removed and we should use `rpc-port`. This pr take the approach of use the `ws-port` set on config as `rpc-port` for the new version of the cli args since I think most of the people that set the `ws port` in their config want to use it to interact with the node.

Also, worth noting that this feature add some overhead to calculate the `cli args version`. I use a map to try to reduce the overhead but in a follow-up pr we can include a config key to allow used to set manually and remove the overhead. 

Thanks!

### CODE REVIEW

File: javascript/packages/orchestrator/src/cmdGenerator.ts [hunk 1]
```diff
@@ -331,16 +321,31 @@ export async function genCmd(
...
-// helper
-// const parachainCollators: ParachainCollatorsInterface =
-//   {} as ParachainCollatorsInterface;
+const getPortFlagsByCliArgsVersion = (nodeSetup: Node) => {
+  // port flags logic
+  const portFlags: { [key: string]: string } = {
+    "--prometheus-port": (nodeSetup.prometheusPort
+      ? nodeSetup.prometheusPort
+      : PROMETHEUS_PORT
+    ).toString(),
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:330]:
```suggestion
    "--prometheus-port": (nodeSetup.prometheusPort || PROMETHEUS_PORT).toString(),
```

File: javascript/packages/orchestrator/src/cmdGenerator.ts [hunk 2]
```diff
@@ -331,16 +321,31 @@ export async function genCmd(
...
 
-// Commented out as "Never used"
-// function getCollatorIndex(paraId: number): number {
-//   if (parachainCollators[paraId] >= 0)
-//     parachainCollators[paraId] = parachainCollators[paraId] + 1;
-//   else parachainCollators[paraId] = 0;
+  if (nodeSetup.substrateCliArgsVersion === SubstrateCliArgsVersion.V0) {
+    portFlags["--rpc-port"] = (
+      nodeSetup.rpcPort ? nodeSetup.rpcPort : RPC_HTTP_PORT
+    ).toString();
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:336]:
```suggestion
    portFlags["--rpc-port"] = (nodeSetup.rpcPort || RPC_HTTP_PORT).toString();
```

File: javascript/packages/orchestrator/src/cmdGenerator.ts [hunk 3]
```diff
@@ -331,16 +321,31 @@ export async function genCmd(
...
+    portFlags["--ws-port"] = (
+      nodeSetup.wsPort ? nodeSetup.wsPort : RPC_WS_PORT
+    ).toString();
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:339]:
```suggestion
    portFlags["--ws-port"] = (nodeSetup.wsPort || RPC_WS_PORT).toString();
```

File: javascript/packages/orchestrator/src/cmdGenerator.ts [hunk 4]
```diff
@@ -331,16 +321,31 @@ export async function genCmd(
...
+  } else {
+    // use ws port as default
+    const portToUse = nodeSetup.wsPort
+      ? nodeSetup.wsPort
+      : nodeSetup.rpcPort
+      ? nodeSetup.rpcPort
+      : RPC_HTTP_PORT;
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:346]:
```suggestion
      : nodeSetup.rpcPort || RPC_HTTP_PORT;
```

File: javascript/packages/orchestrator/src/providers/k8s/substrateCliArgsHelper.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    "", // don't used
+    fullCmd,
+    false,
+  );
+
+  const podDef = await genNodeDef(client.namespace, node);
+  const podName = podDef.metadata.name;
+  await client.spawnFromDef(podDef);
+  const logs = await client.getNodeLogs(podName);
+  debug("logs:");
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/providers/k8s/substrateCliArgsHelper.ts:27]: is this needed?

**Review by wirednkod — APPROVED** [review]
minor corrections

File: javascript/packages/orchestrator/src/cmdGenerator.ts (hunk 1 above)
Comment (l0r1s) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:330]: To simplify:

```suggestion
    "--prometheus-port": (nodeSetup.prometheusPort || PROMETHEUS_PORT).toString(),
```

File: javascript/packages/orchestrator/src/cmdGenerator.ts (hunk 3 above)
Comment (l0r1s) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:339]: Same here:

```suggestion
    portFlags["--rpc-port"] = (nodeSetup.rpcPort || RPC_HTTP_PORT).toString();
    portFlags["--ws-port"] = (nodeSetup.wsPort || RPC_WS_PORT).toString();
```

File: javascript/packages/orchestrator/src/cmdGenerator.ts (hunk 4 above)
Comment (l0r1s) [inline javascript/packages/orchestrator/src/cmdGenerator.ts:346]:
```suggestion
    const portToUse = nodeSetup.wsPort || nodeSetup.rpcPort || RPC_HTTP_PORT;
```

File: javascript/packages/orchestrator/src/providers/k8s/substrateCliArgsHelper.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+          command: collator.command,
+        });
+    }
+    return memo;
+  }, imgCmdMap);
+
+  // check versions in series
+  const promiseGenerators = [];
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for (const [_k, v] of imgCmdMap) {
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/providers/k8s/substrateCliArgsHelper.ts:61]: You can skip values in destructuring:

```suggestion
  for (const [, v] of imgCmdMap) {
```

File: javascript/packages/orchestrator/src/providers/native/substrateCliArgsHelper.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          command: collator.command,
+        });
+    }
+    return memo;
+  }, imgCmdMap);
+
+  // check versions in series
+  const promiseGenerators = [];
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for (const [_k, v] of imgCmdMap) {
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/providers/native/substrateCliArgsHelper.ts:49]:
```suggestion
  for (const [, v] of imgCmdMap) {
```

File: javascript/packages/orchestrator/src/providers/podman/substrateCliArgsHelper.ts [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+          command: collator.command,
+        });
+    }
+    return memo;
+  }, imgCmdMap);
+
+  // check versions in series
+  const promiseGenerators = [];
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for (const [_k, v] of imgCmdMap) {
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/providers/podman/substrateCliArgsHelper.ts:59]:
```suggestion
  for (const [, v] of imgCmdMap) {
```

**Review by l0r1s — COMMENTED** [review]
Some remarks to simplify code but lgtm :+1:

**Pull request merged**